specs/deployment.md

```diff
@@ -1,599 +1,166 @@
-compatibility_date = "2024-10-01"
-
-[vars]
-GITHUB_ALLOWED_USER_ID = "12345678"  # Your GitHub user ID
-
-[[r2_buckets]]
-binding = "STORAGE"
-bucket_name = "second-brain"
-
-[[kv_namespaces]]
-binding = "OAUTH_KV"
-id = "abc123..."  # From step 3
-
-[[kv_namespaces]]
-binding = "RATE_LIMIT_KV"
-id = "def456..."  # From step 3
-
-[triggers]
-crons = ["0 2 * * *"]
-
-[env.development]
-vars = { GITHUB_ALLOWED_USER_ID = "12345678" }
-
-[[env.development.r2_buckets]]
-binding = "STORAGE"
-bucket_name = "second-brain-dev"
-
-[[env.development.kv_namespaces]]
-binding = "OAUTH_KV"
-id = "dev-abc123..."
-
-[[env.development.kv_namespaces]]
-binding = "RATE_LIMIT_KV"
-id = "dev-def456..."
-```
-
-### 5. Set Production Secrets
-
-```bash
-# GitHub OAuth
-wrangler secret put GITHUB_CLIENT_ID
-# Paste your GitHub OAuth App client ID
-
-wrangler secret put GITHUB_CLIENT_SECRET
-# Paste your GitHub OAuth App client secret
-
-# Cookie encryption (generate random 32-byte hex)
-openssl rand -hex 32
-wrangler secret put COOKIE_ENCRYPTION_KEY
-# Paste the generated hex string
-
-# AWS S3 Backup (if using)
-wrangler secret put S3_BACKUP_ACCESS_KEY
-# Paste AWS access key
-
-wrangler secret put S3_BACKUP_SECRET_KEY
-# Paste AWS secret key
-
-wrangler secret put S3_BACKUP_BUCKET
-# Paste S3 bucket name (e.g., my-second-brain-backup)
-
-wrangler secret put S3_BACKUP_REGION
-# Paste AWS region (e.g., us-east-1)
-```
-
-### 6. Set Development Secrets (Optional)
+**Storage isolation** requires separate R2 buckets for user file storage, ensuring development testing cannot corrupt or expose production user data. Development buckets SHOULD use naming conventions that prevent accidental production operations (e.g., `second-brain-dev` vs `second-brain`).
 
-```bash
-# Same as production but for dev environment
-wrangler secret put GITHUB_CLIENT_ID --env development
-wrangler secret put GITHUB_CLIENT_SECRET --env development
-wrangler secret put COOKIE_ENCRYPTION_KEY --env development
-# ... etc
-```
+**Session isolation** requires separate Durable Object namespaces, preventing development session IDs from colliding with production sessions. Each environment's Durable Objects MUST use distinct class names or namespace bindings.
 
----
-
-## Deployment
+**Authentication isolation** requires separate KV namespaces for OAuth tokens and rate limiting state. Development and production MUST use different OAuth client credentials to prevent token reuse across environments.
 
-### Run Tests
+**Analytics isolation** SHOULD separate development and production telemetry to prevent test traffic from skewing production metrics. Cloudflare Analytics Engine SHOULD be configured with environment-specific dataset names.
 
-```bash
-pnpm test
-```
+### Environment Parity
 
-Ensure all tests pass before deploying.
+While isolated, both environments MUST maintain configuration parity to ensure changes validated in development behave identically in production. Parity applies to:
```

Comment:
This is generally true but we need to be able to change config in dev during testing before we ultimately do it in prod. So they should be considered eventually consistent but they shouldn't be expected to be the same at all times.